src/Toolkit/Toolkit/UI/Controls/BasemapGallery/BasemapGalleryController.cs

```diff
@@ -197,24 +197,36 @@ private async Task HandlePortalChanged()
             await UpdateBasemaps();
         }
 
+        private readonly SemaphoreSlim _updateBasemapsSemaphore = new(1, 1);
+
         public async Task UpdateBasemaps()
         {
-            IsLoading = true;
             // Cancel any pending load before starting a new one
             _loadCancellationTokenSource?.Cancel();
-            _loadCancellationTokenSource = new CancellationTokenSource();
+
+            await _updateBasemapsSemaphore.WaitAsync();
             try
             {
-                _availableBasemaps = await PopulateBasemaps(_loadCancellationTokenSource.Token);
-                HandleAvailableBasemapsChanged();
-            }
-            catch (Exception ex)
-            {
-                System.Diagnostics.Trace.WriteLine(ex.Message);
+                IsLoading = true;
+                _loadCancellationTokenSource = new CancellationTokenSource();
+                try
+                {
+                    _availableBasemaps = await PopulateBasemaps(_loadCancellationTokenSource.Token);
```

Comment:
Since we are now serializing calls to PopulateBasemaps, does this make the task-gating code in LoadBasemapGalleryItems(...) redundant?  Perhaps that can be simplified.